recipes/02/02_where.md

```diff
@@ -0,0 +1,48 @@
+# 02 Filtering
+
+## Script
+
+```sql
+-- This example will show how to filter server logs 
+-- in real-time using a standard WHERE clause.
+-- The table it uses is backed by the `faker` connector 
+-- (https://github.com/knaufk/flink-faker) which 
+-- generates rows in memory and is convenient for testing
+-- queries. As such, it is an alternative to the built-in 
+-- `datagen` connector for testing or prototyping queries.
+CREATE TABLE server_logs ( 
+    client_ip STRING,
+    client_identity STRING, 
+    userid STRING, 
+    log_time TIMESTAMP(3),
+    request_line STRING, 
+    status_code STRING, 
+    size INT
+) WITH (
+  'connector' = 'faker', 
+  'fields.client_ip.expression' = '#{Internet.publicIpV4Address}',
+  'fields.client_identity.expression' =  '-',
+  'fields.userid.expression' =  '-',
+  'fields.log_time.expression' =  '#{date.past ''15'',''5'',''SECONDS''}',
+  'fields.request_line.expression' = '#{regexify ''(GET|POST|PUT|PATCH){1}''} #{regexify ''(/search\.html|/login\.html|/prod\.html|cart\.html|/order\.html){1}''} #{regexify ''(HTTP/1\.1|HTTP/2|/HTTP/1\.0){1}''}',
+  'fields.status_code.expression' = '#{regexify ''(200|201|204|400|401|403|301){1}''}',
+  'fields.size.expression' = '#{number.numberBetween ''100'',''10000000''}'
+);
+-- You can filter this (unbounded) table of 
```

Comment:
Just a suggestion, but it'd sound simpler to add "which **continuously** generates rows in memory ..." to the previous explanation and get rid of the "(unbounded)" here.

Reply:
I rewrote this.